pkg/controller/service/service_controller_test.go

```diff
@@ -575,6 +585,27 @@ func TestDoesExternalLoadBalancerNeedsUpdate(t *testing.T) {
 			},
 			expectedNeedsUpdate: true,
 		},
+		{
+			testName: "If service is missing finalizer",
+			updateFn: func() {
+				oldSvc = defaultExternalService()
+				newSvc = defaultExternalService()
+				removeFinalizer(newSvc)
```

Comment:
I might be missing something...From `defaultExternalService()` it seems like there is no finalizers initially so that this `removeFinalizer()` is not needed?